cxx/isce3/core/Utilities.h

```diff
@@ -339,17 +341,28 @@ namespace isce3 { namespace core {
         int left = 0;
         int right = array.size() - 1;
         int index;
+
+        long long count = 0;
         while (left <= right) {
             const int middle = static_cast<int>(std::round(0.5 * (left + right)));
+            const auto middle_value = array[middle];
+            if (std::isnan(middle_value)) {
+                throw std::invalid_argument("input array may not contain NaN values");
+            }
             if (left == (right - 1)) {
                 index = left;
                 return index;
             }
-            if (array[middle] <= value) {
+            if (middle_value <= value) {
                 left = middle;
-            } else if (array[middle] > value) {
+            } else {
                 right = middle;
             }
+            count++;
+            if (count >= array.size()) {
```

Comment:
This will fail if the number of iterations is exactly equal to the array size. For example, if `array.size()` is 1, an exception will always be raised here.

```suggestion
            if (count > array.size()) {
```